pkg/operator/controller/credentials-secret/credentials_secret.go

```diff
@@ -146,6 +147,26 @@ func desiredCredentialsSecret(sourceSecret *corev1.Secret, destName types.Namesp
 				return nil, fmt.Errorf("invalid secret for aws credentials")
 			}
 		}
+
+	case operatorv1beta1.ProviderTypeInfoblox:
+		if username, exists := sourceSecret.Data["EXTERNAL_DNS_INFOBLOX_WAPI_USERNAME"]; !exists || len(username) == 0 {
+			return nil, fmt.Errorf("username not found for infoblox")
+		}
+		if password, exists := sourceSecret.Data["EXTERNAL_DNS_INFOBLOX_WAPI_PASSWORD"]; !exists || len(password) == 0 {
+			return nil, fmt.Errorf("password not found for infoblox")
+		}
+	case operatorv1beta1.ProviderTypeAzure:
+		if creds, exists := secret.Data["azure.json"]; !exists || len(creds) == 0 {
```

Comment:
```suggestion
		if config, exists := sourceSecret.Data["azure.json"]; !exists || len(config) == 0 {
```
`sourceSecret` is better to be used: to be coherent with the rest of the code and it looks more logical too.